File: Python/Programmars/Level_2/예상 대진표.py
from math import ceil


def solution(n, a, b):
    for i in range(1, n):
        if ceil(a/2**i) == ceil(b/2**i):
            return i


print(solution(8, 4, 7))  # 3


# ref
def solution(n, a, b):
    answer = 0
    while a != b:
        answer += 1
        a, b = (a+1)//2, (b+1)//2

    return answer

File: JavaScript/Programmars/Level_2/예상 대진표.js
// https://programmers.co.kr/learn/courses/30/lessons/12985?language=javascript
const solution = (n, a, b) => {
  let answer = 0; // 라운드 카운트

  while (a !== b) {
    a = Math.ceil(a / 2);
    b = Math.ceil(b / 2);
    answer++;
  }

  return answer;
};

console.log(solution(8, 4, 7)); // 3
